galley/pkg/runtime/processing/processor.go

```diff
@@ -0,0 +1,41 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain accumulator copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package processing
+
+import "istio.io/istio/galley/pkg/runtime/resource"
+
+// TODO(nmittler): This interface is a horrific stop-gap that affords the ability to add additional "state"
+//  objects to the existing processor. This entire interface should be refactored or removed in the future.
+
+// Processor of resource events.
+type Processor interface {
+	Handler
+
+	// OnFullSync called by the runtime to indicate that the source has fully synced.
+	OnFullSync()
+
+	// GetSchema returns the schema of types handled by this processor.
+	GetSchema() *resource.Schema
```

Comment:
This is probably not the right abstraction. What is needed here is the input type-list. The original code solved this problem by having functions that do the build of the graph parts.